request-yo-racks/web
# Distinguish env differences in client

Description
-----------
This allows for specifying different config files for client-side code depending whether you're in a development vs. production environment.

Motivation and Context
----------------------
As this app grows, there will inevitably be more and more differences between production and development environments. Having a defined place to put these values will help maintainability.

How Has This Been Tested?
-------------------------
Here's how you can test these changes.
+ Delete `src/config.js`
+ Add `export API_BASE_URL=http://api.192.168.99.100.nip.io/` to `./config/ryr/ryr-env.sh`
+ `make` inside the `web` repository
+ Confirm `src/config.js` gets built and contains a line for `API_BASE_URL`
+ Run the app with `polymer serve` and make sure everything works.

Assuming production also has the environment variable `API_BASE_URL`, this will work there too.

Types of changes
----------------
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)
- Code cleanup / Refactoring

Checklist:
----------
-  [] I have updated the documentation accordingly.

Fixes: https://github.com/request-yo-racks/web/issues/20
